### PR TITLE
chore: use document id instead of id by default where we can

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Describe the technical changes you did.

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Component/Repeatable.tsx [hunk 1]
```diff
@@ -393,8 +393,8 @@ const Component = ({
   isOpen,
   name,
   mainField = {
-    name: 'id',
-    type: 'integer',
+    name: 'documentId',
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Component/Repeatable.tsx:396]: do components have a documentId?

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/DynamicComponent.tsx
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/DynamicComponent.tsx:1]: same here?

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Relations.tsx [hunk 2]
```diff
@@ -158,13 +158,15 @@ const RelationsField = React.forwardRef<HTMLDivElement, RelationsFieldProps>(
...
+
     /**
      * Items that are already connected, but reordered would be in
      * this list, so to get an accurate figure, we remove them.
      */
     const relationsConnected =
       (field.value?.connect ?? []).filter(
-        (rel: Relation) => data.results.findIndex((relation) => relation.id === rel.id) === -1
+        (rel: Relation) =>
+          data.results.findIndex((relation) => relation.documentId === rel.documentId) === -1
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Relations.tsx:169]: we need to find them by `id`

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Component/Repeatable.tsx (hunk 1 above)
Comment (alexandrebodin) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Component/Repeatable.tsx:396]: compos only have ids

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/DynamicComponent.tsx [hunk 3]
```diff
@@ -62,11 +62,11 @@ const DynamicComponent = ({
   } = useDocLayout();
 
   const title = React.useMemo(() => {
-    const { mainField } = components[componentUid]?.settings ?? { mainField: 'id' };
+    const { mainField } = components[componentUid]?.settings ?? { mainField: 'documentId' };
```
Comment (alexandrebodin) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/DynamicComponent.tsx:65]: compose only have id

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Relations.tsx (hunk 2 above)
Comment (alexandrebodin) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/Relations.tsx:169]: documentId is not unique

**Pull request closed without merging**